mailroom part 2 modif

stefanlund opened the pull request:
Woke up with the realization I had introduced an ambiguous way of making a call to start the mailroom_part2. Changed it, looks better.
 
Although it didn't crash or create a bug but under the right circumstances who knows what might have happened.

natasha-aleksandrova commented:
@stefanlund I'm a little confused what you are submitting here?

Title says mailroom part 2 modification but it includes a lot of different files.

Is this your first time submission? Do you have submissions matching this PR in the course website as well?

stefanlund commented:
I apologize for the confusion. I think I need some guidance to sort it out.

I have made some pull requests. Starting out with the git and GitHub, my
intention may not be what you can see. Folder: Stefan_lund, sub-folders:
Lesson_n etc with the homework scripts for each lesson n. There should be
pull requests for Lesson_2, Lesson_3 and Lesson_4 so far.

My (mis)understanding, I have so far only used GitHub. No PR have been made
on the course website.



On Tue, Oct 30, 2018 at 8:48 AM Natasha <notifications@github.com> wrote:

> @stefanlund <https://github.com/stefanlund> I'm a little confused what
> you are submitting here?
>
> Title says mailroom part 2 modification but it includes a lot of different
> files.
>
> Is this your first time submission? Do you have submissions matching this
> PR in the course website as well?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/UWPCE-PythonCert-ClassRepos/Self_Paced-Online/pull/1094#issuecomment-434353684>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AfjDEBWplA3lxV9oJT2DlxmIFmpmZE-bks5uqHTSgaJpZM4X_rgL>
> .
>

natasha-aleksandrova commented:
@stefanlund this here is a PR (Pull Request). So your intention is to submit all of the lessons.

Go to the course website and scroll to the bottom of the page on the assignments, you will see a submission area. Paste PR link for each respective assignment that you included here.

A PR link can be copied from the browser. This PR link is:
`https://github.com/UWPCE-PythonCert-ClassRepos/Self_Paced-Online/pull/1094`

stefanlund commented:
Submitted a link for assignments to lesson 2. Did I do it right?

On Tue, Oct 30, 2018 at 11:50 AM Natasha <notifications@github.com> wrote:

> @stefanlund <https://github.com/stefanlund> this here is a PR (Pull
> Request). So your intention is to submit all of the lessons.
>
> Go to the course website and scroll to the bottom of the page on the
> assignments, you will see a submission area. Paste PR link for each
> respective assignment that you included here.
>
> A PR link can be copied from the browser. This PR link is:
> https://github.com/UWPCE-PythonCert-ClassRepos/Self_Paced-Online/pull/1094
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/UWPCE-PythonCert-ClassRepos/Self_Paced-Online/pull/1094#issuecomment-434422033>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AfjDEAletmLAYOHKMfwaxq9QSeaJRXbtks5uqJ-IgaJpZM4X_rgL>
> .
>

stefanlund commented:
Not sure what the "Files Changed" on the GitHub site does.

Submitted the fizz_buzz exercise. Would appreciate feedback. Unsure if I'm
doing it right.



On Tue, Oct 30, 2018 at 1:26 PM Natasha <notifications@github.com> wrote:

> Yes. Make sure to make that submission for each assignment you included
> here. Check "Files Changed" tab at the top.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/UWPCE-PythonCert-ClassRepos/Self_Paced-Online/pull/1094#issuecomment-434456555>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AfjDEJ6aKqVQEt9qEtZqyZ8PUlsHg3eCks5uqLXhgaJpZM4X_rgL>
> .
>

natasha-aleksandrova commented:
<img width="1119" alt="screen shot 2018-10-31 at 8 09 12 am" src="https://user-images.githubusercontent.com/1928658/47797849-60d49980-dce4-11e8-8f0e-55928da364a7.png">
 Files Changed tab is to view all of the files you submitted.

Your github submission will get reviewed on average within 2 days of submission.

natasha-aleksandrova commented on students/StefanLund/Lesson_2/fizz_buzz.py:
```diff
@@ -0,0 +1,128 @@
+    elif n % 5 == 0:
+        return("Buzz")
+    else:
+        return(n)
```
I'm not sure why there are so many versions of this but this here is the most common one except you would include the for loop over `range` function.

natasha-aleksandrova commented on students/StefanLund/Lesson_2/series.py:
```diff
@@ -0,0 +1,55 @@
+    """
+    for i in range(n):
+        assert (fibonacci(i) == sum_series(i, 0, 1)), "If you read this, there is a bug!"
+        assert (lucas(i) == sum_series(i, 2, 1)), "If you read this, there is a bug!"
```
what if both fibonacci and sum_series have the same bug?

a good assert would be to assert on explicit numbers, example:
`assert fibonacci(4) == 5`

natasha-aleksandrova commented on students/stefan_lund/Lesson_2/fizz_buzz.py:
```diff
@@ -0,0 +1,134 @@
+# python3
```
you have two students folders here, one named `stefan_lund` and `StefanLund` please only use one to avoid confusion.

natasha-aleksandrova commented on students/stefan_lund/Lesson_3/list_lab.py:
```diff
@@ -0,0 +1,125 @@
+
+    fruit_list = [ "Apples", "Pears", "Oranges", "Peaches"]
+    questions_str = "{0}/{1}/{0}/{0}".\
+    format("Enter another fruit", "Enter a number, 1 to 4")
```
this should be indented.

natasha-aleksandrova commented on students/stefan_lund/Lesson_3/slicing_lab.py:
```diff
@@ -0,0 +1,22 @@
+a_tuple = (2, 54, 13, 12, 5, 32)
+
+assert exchange_first_last(a_string) == "ghis is a strint"
+assert exchange_first_last(a_tuple) == (32, 54, 13, 12, 5, 2)
```
this assignment is not complete?

natasha-aleksandrova commented on students/stefan_lund/Lesson_3/mailroom.py:
```diff
@@ -0,0 +1,190 @@
+    print("Leaving the Mailroom.")
+
+
+data = ["William Gates, III", 653784.49, 2, "Mark Zuckerberg", 16396.10, 3, "Jeff Bezos", 877.33, 1, "Paul Allen", 708.42, 3]
```
suggested data structure: `data = [("Kat", [243, 34343]),]`

please see guidelines:
https://uwpce-pythoncert.github.io/PythonCertDevel/exercises/mailroom_tutorial.html#choosing-a-data-structure

natasha-aleksandrova commented on students/stefan_lund/Lesson_3/mailroom.py:
```diff
@@ -0,0 +1,190 @@
+    """
+    # data = []
+    start_options = ["start", "Send a Thank You", "Create a Report", "quit"]
+    question = user_input
```
reassigning a function?

natasha-aleksandrova commented on students/stefan_lund/Lesson_3/mailroom.py:
```diff
@@ -0,0 +1,190 @@
+        else:
+            print("We have a bug!")
+            not_finished = False
+    print("Leaving the Mailroom.")
```
suggestion to simplify this entry point, the logic is a little over-complicated.

please review suggestions here:
https://uwpce-pythoncert.github.io/PythonCertDevel/exercises/mailroom_tutorial.html#controlling-main-program-flow

natasha-aleksandrova commented on students/stefan_lund/Lesson_3/mailroom.py:
```diff
@@ -0,0 +1,190 @@
+        else:
+            d3 = data[i + 1] / data[i + 2]
+        line = temp_form.format(d0, d1, d2, d3, nl = name_len, al = amount_len - 1)
+        print(line)
```
I'm not seeing a sorting feature here, please review the instructions.

natasha-aleksandrova commented on students/stefan_lund/Lesson_3/mailroom.py:
```diff
@@ -0,0 +1,190 @@
+
+    # print thank you note after accounting for the gift
+    thank_you_form = "\n\nDear {name},\n\nThank you for the very generous donation of ${amount}\n\nYours Truly,\n"
+    print(thank_you_form.format(name = name, amount = amount))
```
same comment here, this function is a bit over-complicated - can you refactor to simplify? 


If you need some suggestions, I'd be happy to provide a code sample.

natasha-aleksandrova commented:
@stefanlund you also need to make submissions on the course website to actually get the grade. I only saw two submissions, and please ensure to use **PR Link** not commit link.

PR Link:
https://github.com/UWPCE-PythonCert-ClassRepos/Self_Paced-Online/pull/1094 

Copy URL in the browser bar.

stefanlund commented:
committed the slicing_lab.py

https://github.com/UWPCE-PythonCert-ClassRepos/Self_Paced-Online/pull/1124/files

I cannot find the "Files Changed" button. What am I doing backwards?

On Wed, Oct 31, 2018 at 1:28 PM Stefan <tshlund@gmail.com> wrote:

> Slicing exercise, I will review and submit through the UW website.
> However, in general, I'm somewhat concerned about your comment that you
> only see a partial code. Are other codes possibly also not up to date
> because of my unintended abuse? I feel very unsure navigating the GitHub
> website still. The fact I'm using 2 student folders might explain some of
> it. I made one and deleted it (apparently unsuccessfully but escaped me
> because of how the sorting of upper and lower case is done) and replaced it
> with a new one.
>
>
> Duplicate student folders. How to get rid of the duplicate? Use
>
> $git branch -d 'StefanLund'  or  $rmdir 'StefanLund'  or possibly  $git
> rm 'StefanLund' ?
>
>
> mailroom part 2, sorry about that. It grew to something bigger and
> different than what I set out to accomplish. I had the initial idea of
> creating a generic choice menu object where option_a, option_b etc would be
> displayed as 1 - option_a, 2 - option_b etc where the menu would display
> mailroom options but today could just as well display 1 - scarymovie_a, 2 - scarymovie_b
> etc. I only saw the option to select an integer from the menu to make the
> selection but the whole thing came crashing down when I got to the part of
> creating a list or enter a name … not selecting a number. Enough. I will
> revise.
>
> On GitHub, my intention now is to commit one script only and then submit
> it on the UW website. Where can I find the final script, not indicating
> changes, to submit?
>
> Give me a few days and I will be re-submitting and have more help
> requests. Very much like the program. I have a lot of fun.
>
> On Wed, Oct 31, 2018 at 8:38 AM Natasha <notifications@github.com> wrote:
>
>> *@natasha-aleksandrova* approved this pull request.
>>
>> @stefanlund <https://github.com/stefanlund> alright left some comments
>> for you.
>>
>> assignments that will be graded:
>> grid printer
>> fizz buzz
>> series
>> list lab
>> str format lab
>>
>> assignments that need work:
>> slicing - I see partial code
>> lesson4 / mailroom - please review instructions on what changes should be
>> added in lesson4, OO mailroom is not until Lesson9.
>>
>> General comments:
>>
>>    - please only use one student folder (you currently have duplicated
>>    commits under two different folders)
>>    - please try make your Pull Request with small iterations. It is a
>>    lot to review at once and to provide a lot of detailed feedback which would
>>    also be overwhelming for you as well.
>>    - Great start on the first version of the mailroom however I would
>>    like to see a revision where code is a bit more tightened up with a cleaner
>>    loops logic. Review the links I posted to guidelines, in addition to that
>>    I'm also happy to post some code samples if you would like to see that.
>>
>> Please let me know if you have any additional questions!
>> ------------------------------
>>
>> In students/StefanLund/Lesson_2/fizz_buzz.py
>> <https://github.com/UWPCE-PythonCert-ClassRepos/Self_Paced-Online/pull/1094#discussion_r229737931>
>> :
>>
>> > +def fizz_buzz_1(n):
>> +    """
>> +    returns Fizz if n is a multiple of 3
>> +    returns Buzz if n is a multiple of 5
>> +    returns FizzBuzz if n is a multiple of 3 and 5
>> +    returns n otherwise
>> +    """
>> +
>> +    if n % 3 == 0 and n % 5 == 0:
>> +        return("FizzBuzz")
>> +    elif n % 3 == 0:
>> +        return("Fizz")
>> +    elif n % 5 == 0:
>> +        return("Buzz")
>> +    else:
>> +        return(n)
>>
>> I'm not sure why there are so many versions of this but this here is the
>> most common one except you would include the for loop over range
>> function.
>> ------------------------------
>>
>> In students/StefanLund/Lesson_2/series.py
>> <https://github.com/UWPCE-PythonCert-ClassRepos/Self_Paced-Online/pull/1094#discussion_r229738708>
>> :
>>
>> > +        return i
>> +    elif n == 1:
>> +        return j
>> +    else:
>> +        return sum_series(n - 1, i, j) + sum_series(n - 2, i, j)
>> +
>> +
>> +def test_sum_series(n):
>> +    """
>> +    n: test values in the range 0 to n - 1
>> +    test if there are any dicrepancies in fiboncci and lucas series
>> +    and the sum_series function.
>> +    """
>> +    for i in range(n):
>> +        assert (fibonacci(i) == sum_series(i, 0, 1)), "If you read this, there is a bug!"
>> +        assert (lucas(i) == sum_series(i, 2, 1)), "If you read this, there is a bug!"
>>
>> what if both fibonacci and sum_series have the same bug?
>>
>> a good assert would be to assert on explicit numbers, example:
>> assert fibonacci(4) == 5
>> ------------------------------
>>
>> In students/stefan_lund/Lesson_2/fizz_buzz.py
>> <https://github.com/UWPCE-PythonCert-ClassRepos/Self_Paced-Online/pull/1094#discussion_r229739397>
>> :
>>
>> > @@ -0,0 +1,134 @@
>> +# python3
>>
>> you have two students folders here, one named stefan_lund and StefanLund
>> please only use one to avoid confusion.
>> ------------------------------
>>
>> In students/stefan_lund/Lesson_3/list_lab.py
>> <https://github.com/UWPCE-PythonCert-ClassRepos/Self_Paced-Online/pull/1094#discussion_r229739766>
>> :
>>
>> > +    """
>> +
>> +    """
>> +    prompt = q + " : "
>> +    response = input(prompt)
>> +    return response
>> +
>> +
>> +def series_1():
>> +    """
>> +
>> +    """
>> +
>> +    fruit_list = [ "Apples", "Pears", "Oranges", "Peaches"]
>> +    questions_str = "{0}/{1}/{0}/{0}".\
>> +    format("Enter another fruit", "Enter a number, 1 to 4")
>>
>> this should be indented.
>> ------------------------------
>>
>> In students/stefan_lund/Lesson_3/slicing_lab.py
>> <https://github.com/UWPCE-PythonCert-ClassRepos/Self_Paced-Online/pull/1094#discussion_r229740111>
>> :
>>
>> > +    """
>> +    seq: expected to be of type string or type tuple
>> +    d:   returns d of the same type as seq
>> +    """
>> +    a = seq[-1:]
>> +    b = seq[1:-1]
>> +    c = seq[:1]
>> +    d = a + b + c
>> +    return d
>> +
>> +
>> +a_string = "this is a string"
>> +a_tuple = (2, 54, 13, 12, 5, 32)
>> +
>> +assert exchange_first_last(a_string) == "ghis is a strint"
>> +assert exchange_first_last(a_tuple) == (32, 54, 13, 12, 5, 2)
>>
>> this assignment is not complete?
>> ------------------------------
>>
>> In students/stefan_lund/Lesson_3/mailroom.py
>> <https://github.com/UWPCE-PythonCert-ClassRepos/Self_Paced-Online/pull/1094#discussion_r229741449>
>> :
>>
>> > +                answer = answer.lower()
>> +                wrong_answer = False
>> +
>> +        if answer == "a":
>> +            thank_you(data)
>> +        elif answer == "b":
>> +            report(data)
>> +        elif answer == "c":
>> +            not_finished = False
>> +        else:
>> +            print("We have a bug!")
>> +            not_finished = False
>> +    print("Leaving the Mailroom.")
>> +
>> +
>> +data = ["William Gates, III", 653784.49, 2, "Mark Zuckerberg", 16396.10, 3, "Jeff Bezos", 877.33, 1, "Paul Allen", 708.42, 3]
>>
>> suggested data structure: data = [("Kat", [243, 34343]),]
>>
>> please see guidelines:
>>
>> https://uwpce-pythoncert.github.io/PythonCertDevel/exercises/mailroom_tutorial.html#choosing-a-data-structure
>> ------------------------------
>>
>> In students/stefan_lund/Lesson_3/mailroom.py
>> <https://github.com/UWPCE-PythonCert-ClassRepos/Self_Paced-Online/pull/1094#discussion_r229741825>
>> :
>>
>> > +        d2 = data[i + 2]
>> +        if d2 == 0:
>> +            d3 = 0
>> +        else:
>> +            d3 = data[i + 1] / data[i + 2]
>> +        line = temp_form.format(d0, d1, d2, d3, nl = name_len, al = amount_len - 1)
>> +        print(line)
>> +
>> +
>> +def run_mailroom(data):
>> +    """
>> +    data: list with name, total of donations - dddd.cc, number of gift occations
>> +    """
>> +    # data = []
>> +    start_options = ["start", "Send a Thank You", "Create a Report", "quit"]
>> +    question = user_input
>>
>> reassigning a function?
>> ------------------------------
>>
>> In students/stefan_lund/Lesson_3/mailroom.py
>> <https://github.com/UWPCE-PythonCert-ClassRepos/Self_Paced-Online/pull/1094#discussion_r229742223>
>> :
>>
>> > +            answer = question(start_options, data)
>> +            accept = ["start", ["a", "b", "c"]]
>> +            if accept_user_input(answer, accept):
>> +                answer = answer.lower()
>> +                wrong_answer = False
>> +
>> +        if answer == "a":
>> +            thank_you(data)
>> +        elif answer == "b":
>> +            report(data)
>> +        elif answer == "c":
>> +            not_finished = False
>> +        else:
>> +            print("We have a bug!")
>> +            not_finished = False
>> +    print("Leaving the Mailroom.")
>>
>> suggestion to simplify this entry point, the logic is a little
>> over-complicated.
>>
>> please review suggestions here:
>>
>> https://uwpce-pythoncert.github.io/PythonCertDevel/exercises/mailroom_tutorial.html#controlling-main-program-flow
>> ------------------------------
>>
>> In students/stefan_lund/Lesson_3/mailroom.py
>> <https://github.com/UWPCE-PythonCert-ClassRepos/Self_Paced-Online/pull/1094#discussion_r229742696>
>> :
>>
>> > +    temp_form = "\n{:<{nl}}|{:>{al}}|{:>{al}}|{:>{al}}"
>> +    header = temp_form.format(*temp_str, nl = name_len, al = amount_len)
>> +    print(header)
>> +    print("-" * (name_len + 3 * amount_len + 3))
>> +
>> +    for i in range(0, len(data), 3):
>> +        temp_form = "{:<{nl}} ${:>{al}.2f}  {:>{al}} ${:>{al}.2f}"
>> +        d0 = data[i]
>> +        d1 = data[i + 1]
>> +        d2 = data[i + 2]
>> +        if d2 == 0:
>> +            d3 = 0
>> +        else:
>> +            d3 = data[i + 1] / data[i + 2]
>> +        line = temp_form.format(d0, d1, d2, d3, nl = name_len, al = amount_len - 1)
>> +        print(line)
>>
>> I'm not seeing a sorting feature here, please review the instructions.
>> ------------------------------
>>
>> In students/stefan_lund/Lesson_3/mailroom.py
>> <https://github.com/UWPCE-PythonCert-ClassRepos/Self_Paced-Online/pull/1094#discussion_r229743158>
>> :
>>
>> > +    amount_options = ["amount", "dollars or dollars.cents: "]
>> +    accept = ["amount"]
>> +    while wrong_answer:
>> +        answer = user_input(amount_options, data)
>> +        if accept_user_input(answer, accept):
>> +            wrong_answer = False
>> +            amount = answer
>> +            if "." not in amount:
>> +                amount += '.00'
>> +            amount = round(float(amount), 2)
>> +            data[name_ind + 1] += amount
>> +            data[name_ind + 2] += 1
>> +
>> +    # print thank you note after accounting for the gift
>> +    thank_you_form = "\n\nDear {name},\n\nThank you for the very generous donation of ${amount}\n\nYours Truly,\n"
>> +    print(thank_you_form.format(name = name, amount = amount))
>>
>> same comment here, this function is a bit over-complicated - can you
>> refactor to simplify?
>>
>> If you need some suggestions, I'd be happy to provide a code sample.
>>
>> —
>> You are receiving this because you were mentioned.
>> Reply to this email directly, view it on GitHub
>> <https://github.com/UWPCE-PythonCert-ClassRepos/Self_Paced-Online/pull/1094#pullrequestreview-170307811>,
>> or mute the thread
>> <https://github.com/notifications/unsubscribe-auth/AfjDEOYT3hAp6_ESUfS0IZ4XvFRnadQ1ks5uqcQHgaJpZM4X_rgL>
>> .
>>
>
